### PR TITLE
fix: Improving the ui for when "additionalProperties": true

### LINKED ISSUE #4535: New adaptive forms designs don't adapt well to objects with "additionalProperties": true
The gray box with icons shouldn't be there in objects with "additionalProperties": true.

image

### PR DESCRIPTION
**Pull request by LouisEugeneMSFT**
## Description

The new adaptive forms UI doesn't adapt well to objects with **"additionalProperties": true**: the gray box with icon shouldn't be there in nested ui. This is the case for the **Microsoft.HttpRequest** **body**'s property for example.

This PR addresses this issue (see screenshots of before and after).

## Task Item

closes #4535 

## Screenshots

Before (See **body** property):
image

Before (When adding key value pairs):
image

After: 
image

### CODE REVIEW

File: Composer/packages/adaptive-form/src/components/fields/OneOfField/utils.ts [hunk 1]
```diff
@@ -29,9 +29,9 @@ export function getOptions(
   schema: JSONSchema7,
   definitions?: SchemaDefinitions
 ): { options: IDropdownOption[]; isNested: boolean } {
-  const { type, oneOf } = schema;
+  const { type, oneOf, additionalProperties } = schema;
 
-  let isNested = false;
+  let isNested = additionalProperties ? true : false;
```
Comment (tdurnford) [inline Composer/packages/adaptive-form/src/components/fields/OneOfField/utils.ts:34]: nit
```suggestion
  let isNested = !!additionalProperties;
```

File: Composer/packages/adaptive-form/src/components/fields/OpenObjectField/ObjectItem.tsx [hunk 2]
```diff
@@ -91,6 +91,7 @@ const ObjectItem: React.FC<ObjectItemProps> = ({
             {...rest}
             definitions={definitions}
             id={`${name}.value`}
+            label={false}
```
Comment (tdurnford) [inline Composer/packages/adaptive-form/src/components/fields/OpenObjectField/ObjectItem.tsx:94]: Does this effect other object items? I think it should be based on if the fields are inline or not.

File: Composer/packages/adaptive-form/src/utils/getFieldIconText.ts [hunk 3]
```diff
@@ -16,5 +16,5 @@ export const getFieldIconText = (type: any): string => {
   } else if (typeFormatted === 'expression') {
     return '=';
   }
-  return '';
+  return undefined;
```
Comment (tdurnford) [inline Composer/packages/adaptive-form/src/utils/getFieldIconText.ts:19]: nit: We can remove this.
```suggestion
```

File: Composer/packages/adaptive-form/src/components/fields/OpenObjectField/ObjectItem.tsx (hunk 2 above)
Comment (LouisEugeneMSFT) [reply Composer/packages/adaptive-form/src/components/fields/OpenObjectField/ObjectItem.tsx:94]: **ObjectItem** is only used in **OpenObjectField** and this component already has headers (highlighted code in screenshot below):

image

**Review by tdurnford — APPROVED** [review]

**Review by tdurnford — APPROVED** [review]

**Review by GeoffCoxMSFT — APPROVED** [review]

**Review by GeoffCoxMSFT — APPROVED** [review]

**Pull request merged**